internal/namespaces/baremetal/v1alpha1/custom_server_create_test.go

```diff
@@ -0,0 +1,119 @@
+	baremetal "github.com/scaleway/scaleway-sdk-go/api/baremetal/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+// All test below should succeed to create an instance.
+func Test_CreateServer(t *testing.T) {
+	// Simple use cases
+	t.Run("Simple", func(t *testing.T) {
+		t.Run("Default", core.Test(&core.TestConfig{
+			Commands: GetCommands(),
+			Cmd:      "scw baremetal server create zone=fr-par-2",
+			Check: core.TestCheckCombine(
+				core.TestCheckGolden(),
+				core.TestCheckExitCode(0),
+			),
+			AfterFunc: func(ctx *core.AfterFuncCtx) error {
+				_, err := baremetal.NewAPI(ctx.Client).WaitForServer(&baremetal.WaitForServerRequest{
+					ServerID: ctx.CmdResult.(*baremetal.Server).ID,
+				})
+				if err != nil {
+					return err
+				}
+				_, err = baremetal.NewAPI(ctx.Client).DeleteServer(&baremetal.DeleteServerRequest{
+					ServerID: ctx.CmdResult.(*baremetal.Server).ID,
+				})
+				if err != nil {
+					return err
+				}
+				return nil
+			},
+			DefaultZone: scw.ZoneFrPar2,
+		}))
+
+		t.Run("With name", core.Test(&core.TestConfig{
+			Commands: GetCommands(),
+			Cmd:      "scw baremetal server create name=test-create-server-with-name zone=fr-par-2",
+			Check: core.TestCheckCombine(
+				func(t *testing.T, ctx *core.CheckFuncCtx) {
+					assert.Equal(t, "test-create-server-with-name", ctx.Result.(*baremetal.Server).Name)
+				},
+				core.TestCheckExitCode(0),
+			),
+			DefaultZone: scw.ZoneFrPar2,
+			AfterFunc: func(ctx *core.AfterFuncCtx) error {
+				_, err := baremetal.NewAPI(ctx.Client).WaitForServer(&baremetal.WaitForServerRequest{
+					ServerID: ctx.CmdResult.(*baremetal.Server).ID,
+				})
+				if err != nil {
+					return err
+				}
+
+				_, err = baremetal.NewAPI(ctx.Client).DeleteServer(&baremetal.DeleteServerRequest{
+					ServerID: ctx.CmdResult.(*baremetal.Server).ID,
+				})
+				if err != nil {
+					return err
+				}
+
+				return nil
+			},
+		}))
+
+		t.Run("Tags", core.Test(&core.TestConfig{
+			Commands: GetCommands(),
+			Cmd:      "scw baremetal server create tags.0=prod tags.1=blue zone=fr-par-2",
+			Check: core.TestCheckCombine(
+				func(t *testing.T, ctx *core.CheckFuncCtx) {
+					assert.Equal(t, "prod", ctx.Result.(*baremetal.Server).Tags[0])
+					assert.Equal(t, "blue", ctx.Result.(*baremetal.Server).Tags[1])
+				},
+				core.TestCheckExitCode(0),
+			),
+			DefaultZone: scw.ZoneFrPar2,
+			AfterFunc: func(ctx *core.AfterFuncCtx) error {
+				_, err := baremetal.NewAPI(ctx.Client).WaitForServer(&baremetal.WaitForServerRequest{
+					ServerID: ctx.CmdResult.(*baremetal.Server).ID,
+				})
+				if err != nil {
+					return err
+				}
+
+				_, err = baremetal.NewAPI(ctx.Client).DeleteServer(&baremetal.DeleteServerRequest{
+					ServerID: ctx.CmdResult.(*baremetal.Server).ID,
+				})
+				if err != nil {
+					return err
+				}
+
+				return nil
+			},
+		}))
+	})
+}
+
+// None of the tests below should succeed to create an instance.
+func Test_CreateServerErrors(t *testing.T) {
```

Comment:
IMHO it is better to keep all test of a single command inside a single function. Consider adding a `t.Run` in the fonction above